### PR TITLE
fix: add redis support [3488]

### PR DESCRIPTION
**Pull request by doriaviram**

### CODE REVIEW

File: example/parent.js [hunk 1]
```diff
@@ -12,9 +12,7 @@ const parentFn = async (event, context, callback) => {
   const AWS = require('aws-sdk');
   AWS.config.update({ region: 'us-east-1' });
   const lambda = new AWS.Lambda();
-  const x = await lambda
-    .invoke({ FunctionName: 'aws-nodejs-dev-child' })
-    .promise();
+  const x = await lambda.invoke({ FunctionName: 'aws-nodejs-dev-child' }).promise();
```
Comment (doriaviram) [inline example/parent.js:15]: ignore

File: src/hooks/redis.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    ended,
+    error,
+    result,
+  });
+  SpansContainer.addSpan(span);
+};
+
+function sendCommandBeforeHook(args) {
+  const command = args[0];
+  if (this.ready === false || this.stream.writable === false || command.command === 'info') {
```
Comment (nirLumigo) [inline src/hooks/redis.js:21]: How do we have `this.read` and `this.stream`? Does `this.stream` always defined?

File: src/hooks/redis.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import { safeRequire } from '../utils/requireUtils';
+import { hook } from '../extender';
+import { getRandomId } from '../utils';
+import { createRedisSpan, extendRedisSpan } from '../spans/redisSpan';
+import { SpansContainer } from '../globals';
+import * as logger from '../logger';
+
+const createCallbackHandler = redisSpan => args => {
+  const ended = new Date();
+  const [error, result] = args;
```
Comment (nirLumigo) [inline src/hooks/redis.js:10]: This function is in the `beforeHook` so how can it get `error` and `result`?
Aren't they only available in the `afterHook`?

File: src/spans/redisSpan.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { getBasicChildSpan, REDIS_SPAN } from './awsSpan';
+import { payloadStringify } from '../utils/payloadStringify';
+
+export const createRedisSpan = (spanId, requestMetadata, redisFields) => {
+  const baseSpan = getBasicChildSpan(spanId, REDIS_SPAN);
+  return {
+    ...baseSpan,
+    started: requestMetadata.started,
+    requestCommand: redisFields.command.command,
```
Comment (nirLumigo) [inline src/spans/redisSpan.js:9]: Is `redisFields.command` never `undefined`?

File: testUtils/redisMocker.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+          command: 'set',
+          args: [key, value],
+        },
+        callback: callback,
+      });
+    },
+  };
+}
+
+//In this implantation we create only 1 client at time
```
Comment (nirLumigo) [inline testUtils/redisMocker.js:18]: Typo: implantation

File: src/hooks/redis.js (hunk 2 above)
Comment (doriaviram) [reply src/hooks/redis.js:21]: Fixed the `this.stream`

File: src/hooks/redis.js (hunk 3 above)
Comment (doriaviram) [reply src/hooks/redis.js:10]: This is the wrapping of the callback, Redis client calls customer callback with: `callback(err, data)`

File: src/spans/redisSpan.js (hunk 4 above)
Comment (doriaviram) [reply src/spans/redisSpan.js]: Fixed

File: testUtils/redisMocker.js (hunk 5 above)
Comment (doriaviram) [reply testUtils/redisMocker.js:18]: Think its the same

**Pull request merged**